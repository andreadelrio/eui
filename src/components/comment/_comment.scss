.euiComment {
  font-size: $euiFontSizeS;
  display: flex;
  margin: -($euiSize * .5);
  padding-bottom: $euiSize;
  min-height: $euiSize * 5;

  .euiCommentEvent {
    flex-grow: 1;
  }

  .euiCommentTimeline {
    position: relative;
    flex-grow: 0;

    &::before {
      content: '';
      position: absolute;
      left: $euiSizeXXL / 2;
      top: $euiSizeXXL;
      width: $euiSizeXS / 2;
      background-color: $euiColorLightShade;
      height: 100%;
    }
  }

  .euiCommentTimeline, .euiCommentEvent {
    flex-basis: auto;
    margin: $euiSize * .5;
  }

}