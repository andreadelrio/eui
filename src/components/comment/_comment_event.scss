@include euiPanel($selector: '.euiCommentEvent--regular');

.euiCommentEvent {
  font-size: $euiFontSizeS;
  overflow: hidden;

  .euiCommentEvent__event {
    padding-right: $euiSizeXS;
  }

  .euiCommentEvent__header {
    @include euiBreakpoint('xs', 's') {
      margin: -($euiSizeXS);

      >div {
        margin: $euiSizeXS;
      }
    }

    align-items: center;

    .euiCommentEvent__headerData>div {
      padding-right: $euiSizeXS;
    }
  }

  .euiCommentEvent__headerData {
    align-items: center;
    display: flex;
    flex-wrap: wrap;
  }

  .euiCommentEvent__headerUsername {
    font-weight: $euiFontWeightSemiBold;
  }

  .euiCommentEvent__body {
    line-height: $euiSize * 1.2;
  }

}

.euiCommentEvent--regular {
  .euiCommentEvent__header {
    background-color: $euiColorLightestShade;
    border-bottom: $euiBorderThin;
    padding: $euiSizeM;
    display: flex;
    justify-content: space-between;
    align-items: center;
  }

  .euiCommentEvent__body {
    padding: $euiSizeS;
  }
}

.euiCommentEvent--update {
  .euiCommentEvent__header {
    padding-top: $euiSizeS;
    display: flex;
    justify-content: flex-start;
  }

  .euiCommentEvent__headerData {
    padding-right: $euiSizeS;
  }

  .euiCommentEvent__body {
    padding-top: $euiSizeS;
  }
}