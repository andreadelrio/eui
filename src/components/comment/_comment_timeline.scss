.euiCommentTimeline__content {
  min-width: $euiSizeXXL;
  min-height: $euiSizeXXL;
  display: flex;
  justify-content: center;
  align-items: center;
  position: relative;
  background-color: $euiColorEmptyShade;
}