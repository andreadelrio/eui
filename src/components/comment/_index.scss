@import 'comment';
@import 'comment_event';
@import 'comment_timeline';
@import 'comment_list';
