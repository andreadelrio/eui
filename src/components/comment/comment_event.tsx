import React, { FunctionComponent, ReactNode } from 'react';
import { CommonProps, keysOf } from '../common';
import classNames from 'classnames';

export type EuiCommentEventProps = CommonProps &
  CommonProps & {
    /**
     * Author of the comment. Display a small icon or avatar with it if needed.
     */
    username: ReactNode;
    timestamp?: ReactNode;
    /**
     * Describes the event that took place
     */
    event?: ReactNode;
    /**
     * Actions the user can perform from the comment's header
     */
    actions?: ReactNode;
    /**
     * Use "update" when the comment is primarily showing info about actions that the user or the system has performed (e.g. "user1 edited a case").
     */
    type?: EuiCommentType;
  };

const typeToClassNameMap = {
  regular: 'euiCommentEvent--regular',
  update: 'euiCommentEvent--update',
};

export const TYPES = keysOf(typeToClassNameMap);
export type EuiCommentType = keyof typeof typeToClassNameMap;

export const EuiCommentEvent: FunctionComponent<EuiCommentEventProps> = ({
  children,
  className,
  username,
  timestamp,
  type = 'regular',
  event,
  actions,
  ...rest
}) => {
  const classes = classNames(
    'euiCommentEvent',
    typeToClassNameMap[type],
    className
  );

  return (
    <div data-focusable="true" className={classes} {...rest}>
      <div className="euiCommentEvent__header">
        <div className="euiCommentEvent__headerData">
          <div className="euiCommentEvent__headerUsername">{username}</div>
          <div className="euiCommentEvent__headerEvent">{event}</div>
          <div className="euiCommentEvent__headerTimestamp">{timestamp}</div>
        </div>
        <div className="euiCommentEvent__headerActions">{actions}</div>
      </div>
      <div className="euiCommentEvent__body">{children}</div>
    </div>
  );
};
