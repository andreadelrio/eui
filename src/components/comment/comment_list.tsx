import React, { HTMLAttributes, FunctionComponent } from 'react';
import { CommonProps } from '../common';
import classNames from 'classnames';

export type EuiCommentListProps = HTMLAttributes<HTMLDivElement> & CommonProps & {

};

export const EuiCommentList: FunctionComponent<EuiCommentListProps> = ({
  children,
  className,
  ...rest
}) => {
  const classes = classNames('euiCommentList', className);

  return (
    <div
      className={classes}
      {...rest}
    >
      {children}
    </div>
  );
};
