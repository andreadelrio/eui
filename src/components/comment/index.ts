export { EuiComment } from './comment';

export { EuiCommentEvent } from './comment_event';

export { EuiCommentTimeline } from './comment_timeline';

export { EuiCommentList } from './comment_list';
