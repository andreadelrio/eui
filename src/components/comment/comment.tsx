import React, { FunctionComponent } from 'react';
import { CommonProps } from '../common';
import classNames from 'classnames';

import { EuiCommentEvent, EuiCommentEventProps } from './comment_event';
import {
  EuiCommentTimeline,
  EuiCommentTimelineProps,
} from './comment_timeline';

export type EuiCommentProps = EuiCommentEventProps &
  EuiCommentTimelineProps &
  CommonProps & {};

export const EuiComment: FunctionComponent<EuiCommentProps> = ({
  children,
  className,
  username,
  event,
  actions,
  timelineIcon,
  type = 'regular',
  timestamp,
  ...rest
}) => {
  const classes = classNames('euiComment', className);

  return (
    <div className={classes} {...rest}>
      <EuiCommentTimeline timelineIcon={timelineIcon} />
      <EuiCommentEvent
        username={username}
        actions={actions}
        event={event}
        timestamp={timestamp}
        type={type}>
        {children}
      </EuiCommentEvent>
    </div>
  );
};
