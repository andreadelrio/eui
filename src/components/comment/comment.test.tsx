import React from 'react';
import { render } from 'enzyme';
import { requiredProps } from '../../test/required_props';

import { EuiComment } from './comment';

describe('EuiComment', () => {
  test('is rendered', () => {
    const component = render(<EuiComment {...requiredProps} />);

    expect(component).toMatchSnapshot();
  });

  describe('props', () => {
    describe('type', () => {
      it('is rendered', () => {
        const component = render(<EuiComment type="update" />);

        expect(component).toMatchSnapshot();
      });
    });
  });
});
