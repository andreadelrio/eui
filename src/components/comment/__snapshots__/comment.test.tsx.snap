// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EuiComment is rendered 1`] = `
<div
  aria-label="aria-label"
  class="euiComment testClass1 testClass2"
  data-test-subj="test subject string"
>
  <div
    class="euiCommentTimeline"
  >
    <div
      class="euiCommentTimeline__content"
    >
      <div
        class="euiCommentTimeline__contentDefault"
      >
        <div
          data-euiicon-type="user"
        />
      </div>
    </div>
  </div>
  <div
    class="euiCommentEvent euiCommentEvent--regular"
    data-focusable="true"
  >
    <div
      class="euiCommentEvent__header"
    >
      <div
        class="euiCommentEvent__headerData"
      >
        <div
          class="euiCommentEvent__headerUsername"
        >
          someuser
        </div>
         
      </div>
      <div
        class="euiCommentEvent__headerActions"
      />
    </div>
    <div
      class="euiCommentEvent__body"
    />
  </div>
</div>
`;

exports[`EuiComment props event is rendered 1`] = `
<div
  class="euiComment"
>
  <div
    class="euiCommentTimeline"
  >
    <div
      class="euiCommentTimeline__content"
    >
      <div
        class="euiCommentTimeline__contentDefault"
      >
        <div
          data-euiicon-type="user"
        />
      </div>
    </div>
  </div>
  <div
    class="euiCommentEvent euiCommentEvent--regular"
    data-focusable="true"
  >
    <div
      class="euiCommentEvent__header"
    >
      <div
        class="euiCommentEvent__headerData"
      >
        <div
          class="euiCommentEvent__headerUsername"
        >
          someuser
        </div>
        commented 
      </div>
      <div
        class="euiCommentEvent__headerActions"
      />
    </div>
    <div
      class="euiCommentEvent__body"
    />
  </div>
</div>
`;

exports[`EuiComment props timelineIcon is rendered 1`] = `
<div
  class="euiComment"
>
  <div
    class="euiCommentTimeline"
  >
    <div
      class="euiCommentTimeline__content"
    >
      <div
        aria-label="Mario"
        class="euiAvatar euiAvatar--l euiAvatar--user"
        style="background-image:none;background-color:#f1d86f;color:#000000"
        title="Mario"
      >
        <span
          aria-hidden="true"
        >
          M
        </span>
      </div>
    </div>
  </div>
  <div
    class="euiCommentEvent euiCommentEvent--regular"
    data-focusable="true"
  >
    <div
      class="euiCommentEvent__header"
    >
      <div
        class="euiCommentEvent__headerData"
      >
        <div
          class="euiCommentEvent__headerUsername"
        >
          someuser
        </div>
         
      </div>
      <div
        class="euiCommentEvent__headerActions"
      />
    </div>
    <div
      class="euiCommentEvent__body"
    />
  </div>
</div>
`;

exports[`EuiComment props timestamp is rendered 1`] = `
<div
  class="euiComment"
>
  <div
    class="euiCommentTimeline"
  >
    <div
      class="euiCommentTimeline__content"
    >
      <div
        class="euiCommentTimeline__contentDefault"
      >
        <div
          data-euiicon-type="user"
        />
      </div>
    </div>
  </div>
  <div
    class="euiCommentEvent euiCommentEvent--regular"
    data-focusable="true"
  >
    <div
      class="euiCommentEvent__header"
    >
      <div
        class="euiCommentEvent__headerData"
      >
        <div
          class="euiCommentEvent__headerUsername"
        >
          someuser
        </div>
         21 days ago
      </div>
      <div
        class="euiCommentEvent__headerActions"
      />
    </div>
    <div
      class="euiCommentEvent__body"
    />
  </div>
</div>
`;

exports[`EuiComment props type is rendered 1`] = `
<div
  class="euiComment"
>
  <div
    class="euiCommentTimeline"
  >
    <div
      class="euiCommentTimeline__content"
    >
      <div
        class="euiCommentTimeline__contentDefault"
      >
        <div
          data-euiicon-type="user"
        />
      </div>
    </div>
  </div>
  <div
    class="euiCommentEvent euiCommentEvent--update"
    data-focusable="true"
  >
    <div
      class="euiCommentEvent__header"
    >
      <div
        class="euiCommentEvent__headerData"
      >
        <div
          class="euiCommentEvent__headerUsername"
        >
          someuser
        </div>
         
      </div>
      <div
        class="euiCommentEvent__headerActions"
      />
    </div>
    <div
      class="euiCommentEvent__body"
    />
  </div>
</div>
`;

exports[`EuiComment renders a body 1`] = `
<div
  class="euiComment"
>
  <div
    class="euiCommentTimeline"
  >
    <div
      class="euiCommentTimeline__content"
    >
      <div
        class="euiCommentTimeline__contentDefault"
      >
        <div
          data-euiicon-type="user"
        />
      </div>
    </div>
  </div>
  <div
    class="euiCommentEvent euiCommentEvent--regular"
    data-focusable="true"
  >
    <div
      class="euiCommentEvent__header"
    >
      <div
        class="euiCommentEvent__headerData"
      >
        <div
          class="euiCommentEvent__headerUsername"
        >
          someuser
        </div>
         
      </div>
      <div
        class="euiCommentEvent__headerActions"
      />
    </div>
    <div
      class="euiCommentEvent__body"
    >
      <p>
        This is the body.
      </p>
    </div>
  </div>
</div>
`;
